src/stream/src/executor/managed_state/top_n/top_n_state.rs

```diff
@@ -61,36 +66,38 @@ impl<S: StateStore> ManagedTopNState<S> {
         self.state_table.delete(value);
     }
 
-    fn get_topn_row(&self, row: Row) -> TopNStateRow {
+    fn get_topn_row(&self, row: Row, group_key_len: usize) -> TopNStateRow {
         let mut datums = Vec::with_capacity(self.state_table.pk_indices().len());
-        for pk_index in self.state_table.pk_indices() {
+        for pk_index in self.state_table.pk_indices().iter().skip(group_key_len) {
             datums.push(row[*pk_index].clone());
         }
```

Comment:
Should the capacity also be updated, or just `collect` it? BTW, it's really non-sense to first collect an owned row, then transform it to an ordered row. 😅

Reply:
The Row is owned by the TopNStateRow :think: